packages/hurumap-core/src/ShareButton/ShareButton.js

```diff
@@ -0,0 +1,78 @@
+import { Box, SvgIcon } from "@mui/material";
+import React from "react";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import {
+  EmailShareButton,
+  FacebookShareButton,
+  FacebookMessengerShareButton,
+  LinkedinShareButton,
+  PinterestShareButton,
+  RedditShareButton,
+  TelegramShareButton,
+  TwitterShareButton,
+  WhatsappShareButton,
+} from "react-share";
+
+const componentMap = {
+  Email: EmailShareButton,
+  Facebook: FacebookShareButton,
+  FacebookMessenger: FacebookMessengerShareButton,
+  LinkedIn: LinkedinShareButton,
+  Pinterest: PinterestShareButton,
+  Reddit: RedditShareButton,
+  Telegram: TelegramShareButton,
+  Twitter: TwitterShareButton,
+  WhatsApp: WhatsappShareButton,
+};
+
+const ShareButton = React.forwardRef(function ShareButton(
+  { ButtonProps, IconProps, icon, name, url, onCopy, ...props },
+  ref,
+) {
+  const SocialButtonComponent = componentMap[name];
+
+  if (!SocialButtonComponent && name !== "CopyUrl") {
+    return null;
+  }
```

Comment:
I think you misunderstood my previous comment; I didn't mean you should add FacebookMessenger button.

```suggestion
  { ButtonProps, IconProps, component, icon, name, onCopy, url, ...props },
  ref,
) {
  let SocialButtonComponent = component;
  if (name && name !== "CopyUrl") {
   SocialButtonComponent = componentMap[name];
  }
  if (!SocialButtonComponent) {
    return null;
  }
```

1. By allowing `component` prop, someone can set it to whatever button they want and we'll use it.
2. If we're copying to clipboard, not sure why name wasn't `Clipboard`. CopyUrl is very implementation specific.